### ISSUE TITLE
Finding it difficult to reproduce fine-tune training on single GPU (24GB)

### ISSUE THREAD

**Issue by MRGLabs** [zihangdai/xlnet#91]
While trying to get a good feel for the performance of the pre-trained model on 'IMDB: movie review sentiment classification', I was hoping to run a simple, reproducible, fine-tuning run with learning_rate=0, train_steps=1, and iterations=1, but have yet to succeed reproducing any loss results.  

**NOTE: I'm running run_classifier.py , and have commented out the two lines of code that shuffle the examples for training - (501, 710).**
```
  if is_training:
     #d = d.shuffle(buffer_size=FLAGS.shuffle_buffer)
  ...
  if FLAGS.do_train:
    ...
    #np.random.shuffle(train_examples)

```
Here is my full configuration:
```
CUDA_VISIBLE_DEVICES=0
--do_train=True
--do_eval=False
--task_name=imdb
--data_dir=data/imdb/aclImdb
--output_dir=proc_data/imdb
--model_dir=exp/imdb
--uncased=False
--spiece_model_file=models/xlnet_cased_L-24_H-1024_A-16/spiece.model
--model_config_path=models/xlnet_cased_L-24_H-1024_A-16/xlnet_config.json
--init_checkpoint=models/xlnet_cased_L-24_H-1024_A-16/xlnet_model.ckpt
--max_seq_length=512
--train_batch_size=4
--eval_batch_size=4
--num_hosts=1
--num_core_per_host=1
--learning_rate=0
--train_steps=1
--warmup_steps=1
--save_steps=500
--iterations=1
--eval_all_ckpt=True
--is_regression=False
--overwrite_data=False
```
Here are some of the loss results:
   ```
    # loss = 1.0395325, step = 0
    # loss = 0.7751261, step = 0
    # loss = 0.6002699, step = 0
    # loss = 1.0697793, step = 0
```

What am I missing here?

**Comment by kimiyoung** [comment]
There are two reasons: (1) a random seed is not used for initialization; (2) Tensorflow will not produce the same results from run to run, even with the same random seed. However, if you finetune the network to convergence, the results should be fine.

**Comment by MRGLabs** [comment]
Zihang, thank you for the response and great code!
